key.core/src/main/antlr4/KeYLexer.g4

```diff
@@ -450,7 +448,7 @@ DOUBLE_LITERAL:
     ;
 
 REAL_LITERAL:
-    RATIONAL_LITERAL ('r' | 'R')
+    RATIONAL_LITERAL ('r' | 'R')?
```

Comment:
`RATIONAL_LITERAL` is a fragment, and therefore does not occur as a standalone token.

`fragment RATIONAL_LITERAL: ...`

Reply:
I would like to have the type suffix mandatory in .key files. "1.4" alone is too ambiguous. If at all, I would say that this would be a double value, where as 1.4r is the real; like it is currently in (our) JML.

So, I still object this change.

Reply:
Sorry, I have to revive this thread: Is the suffix still optional? What was the decision on this? I would agree to @mattulbrich that having floats and reals without mandatory suffix for disambiguation seems dangerous.